### PR TITLE
storage: more descriptive logging for persist_source/shard_source

### PR DESCRIPTION
**Pull request by aljoscha**
To help in debugging!

### Checklist

- [ ] This PR has adequate test coverage / QA involvement has been duly considered.
- [ ] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration or tests, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

**Review by rjobanp — APPROVED** [review]

**Review by danhhz — APPROVED** [review]

File: src/compute/src/sink/persist_sink.rs [hunk 1]
```diff
@@ -107,6 +107,7 @@ where
     let source_as_of = None;
     let (ok_stream, err_stream, token) = mz_storage_operators::persist_source::persist_source(
         &mut desired_collection.scope(),
+        "persist_sink".to_string(),
```
Comment (guswynn) [inline src/compute/src/sink/persist_sink.rs:110]: should this and the storage sink one include the global id?

**Comment by danhhz** [comment]
Uh oh! This seems to be using these reasons as shard names, which is incorrect. We should fix this before the release cut tomorrow

image

**Comment by danhhz** [comment]
Wait, maybe I spoke too soon. @aljoscha Any chance you ran some earlier version of this on staging that used these as names instead of reasons?

**Comment by aljoscha** [comment]
The name eventually makes its way here: https://github.com/MaterializeInc/materialize/blob/31c851df38b2509edd7849f1c4e8759df016d937/src/persist-client/src/operators/shard_source.rs#L269

So yes, it changes the `handle_purpose` and `shard_name`. Reverting... 🙈

**Pull request merged**